src/mapper/pkg/clouduploader/cloud_upload.go

```diff
@@ -0,0 +1,120 @@
+package clouduploader
+
+import (
+	"context"
+	"fmt"
+	"github.com/Khan/genqlient/graphql"
+	"github.com/otterize/network-mapper/src/mapper/pkg/cloudapi"
+	"github.com/otterize/network-mapper/src/mapper/pkg/config"
+	"github.com/otterize/network-mapper/src/mapper/pkg/resolvers"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+	"time"
+)
+
+type Config struct {
+	ClientId       string
+	Secret         string
+	CloudAddress   string
+	Environment    string
+	IntentSource   string
+	UploadInterval int
+}
+
+type CloudUploader struct {
+	intentsHolder       *resolvers.IntentsHolder
+	config              Config
+	tokenSrc            oauth2.TokenSource
+	cloudAPI            cloudapi.CloudAPI
+	lastUploadTimestamp time.Time
+}
+
+func NewCloudUploader(intentsHolder *resolvers.IntentsHolder, config Config, cloudApiImpl cloudapi.CloudAPI) *CloudUploader {
```

Comment:
It's not `cloudApiImpl ` it's just a `cloudApi` (or more accurately it's cloudClient)